docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+### Current State Representation
+
+The cloud backend (or any OTA update system) needs to keep track of the current state of the device. This current state information is relevant as input for the context pipeline for compiling future desired states and acts as an indicator of problems in applying a given desired state, if the current state differs from the desired state, which may not be reached.
+
+For representing the current state of the device, the approach of a holistic inventory of hardware and software nodes will be applied. This model is using a graph structure consisting of three element types - hardware nodes, software nodes, associations.
+
+### Current State Data Model
+
+The following table describes all supported properties and sections of the Current State specification:
+
+| Property | Type | Description |
+| - | - | - |
+| **General properties** | | |
+| hardwareNodes | JSON array | Inventory for a list of hardware nodes |
+| softwareNodes | JSON array | Inventory for a list of software nodes |
+| associations | JSON array | List of mappings between the inventory nodes |
+| **Hardware node properties** | | |
+| id | string | Identifier of the hardware node |
+| version | string | Version of the hardware node |
+| name | string | Name of a hardware node |
+| parameters | JSON array | List of key/value parameters for a hardware node |
+| addressable | boolean | Enables hardware node addressability |
+| **Software node properties** | | |
+| id | string | Identifier of the software node |
+| version | string | Version of the software node |
+| name | string | Name of the software node |
+| parameters | JSON array | List of key/value parameters for a software node |
+| type | string | Type of the software node. The supported types are listed below |
+| **Parameter properties** | | |
+| key | string | Key of the parameter |
+| value | string | Value of the parameter |
+| **Association parameter** | | |
+| sourceId | string | Identifier of the source node of the association |
+| targetId | string | Identifier of the target node of the association |
+
+### Supported software types
+
+The list of the supported software types :
+
+| Type | Description |
+| - | - |
+| IMAGE | Represents an image software type |

```

Comment:
At least for me it would be helpful to specify what kind of image we are talking about here (OS image, container image, etc.)

Reply:
Such questions shall be cleared by the Eclipse BfB project in the future when they put the Inventory specification in the public. 
Currently type IMAGE is used by the self-update agent and denotes the OS image running on the device. 
Containers update agent uses type CONTAINER for its software nodes representing the available containers in the system.